lib/streamlit/elements/widgets/selectbox.py

```diff
@@ -82,53 +112,113 @@ def selectbox(
         key: Key | None = None,
         help: str | None = None,
         on_change: WidgetCallback | None = None,
         args: WidgetArgs | None = None,
         kwargs: WidgetKwargs | None = None,
         *,  # keyword-only arguments:
-        placeholder: str = "Choose an option",
+        placeholder: str = DEFAULT_PLACEHOLDER,
         disabled: bool = False,
         label_visibility: LabelVisibility = "visible",
+        accept_new_options: Literal[False] = False,
     ) -> T: ...
 
+    @overload
+    def selectbox(
+        self,
+        label: str,
+        options: OptionSequence[T],
+        index: int = 0,
+        format_func: Callable[[Any], str] = str,
+        key: Key | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        *,  # keyword-only arguments:
+        placeholder: str = DEFAULT_PLACEHOLDER,
+        disabled: bool = False,
+        label_visibility: LabelVisibility = "visible",
+        accept_new_options: Literal[True] = True,
+    ) -> T | str: ...
+
     @overload
     def selectbox(
         self,
         label: str,
         options: OptionSequence[T],
         index: None,
-        format_func: Callable[[Any], Any] = str,
+        format_func: Callable[[Any], str] = str,
         key: Key | None = None,
         help: str | None = None,
         on_change: WidgetCallback | None = None,
         args: WidgetArgs | None = None,
         kwargs: WidgetKwargs | None = None,
         *,  # keyword-only arguments:
-        placeholder: str = "Choose an option",
+        placeholder: str = DEFAULT_PLACEHOLDER,
         disabled: bool = False,
         label_visibility: LabelVisibility = "visible",
+        accept_new_options: Literal[False] = False,
     ) -> T | None: ...
 
+    @overload
+    def selectbox(
+        self,
+        label: str,
+        options: OptionSequence[T],
+        index: None,
+        format_func: Callable[[Any], str] = str,
+        key: Key | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        *,  # keyword-only arguments:
+        placeholder: str = DEFAULT_PLACEHOLDER,
+        disabled: bool = False,
+        label_visibility: LabelVisibility = "visible",
+        accept_new_options: Literal[True] = True,
+    ) -> T | str | None: ...
+
+    @overload
+    def selectbox(
+        self,
+        label: str,
+        options: OptionSequence[T],
+        index: int | None = 0,
+        format_func: Callable[[Any], str] = str,
+        key: Key | None = None,
+        help: str | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        *,  # keyword-only arguments:
+        placeholder: str = DEFAULT_PLACEHOLDER,
```

Comment:
I think just using `None` here as default might be slightly preferred and more consistent with other similar cases we have where the default is dynamic and not static (that changes depending on other parameters). Having `DEFAULT_PLACEHOLDER` here also might not provide a lot of benefit in our docs or in the IDE help since its probably not translated back to a string (but not 100% sure on that).

Reply:
Oh ok so you mean having `None` here but then still set it to the default one in the body? That makes sense

Reply:
yep 👍

Reply:
Updated it!